D102 energy-function instrumentation, ATM_UTILS.f part: ffdd_te in getTotalEnergy (after GLOBALSUM), ffdd_eadd(1|2) in addEnergyAsDiffuseHeat. Helpers are in ATM_DRV_dynD.f.patch (units 1124, 1125). Generated by diff -u against the PRISTINE ATM_UTILS.f (2026-10-05); local hunks.

--- ATM_UTILS.f.orig
+++ ATM_UTILS.f
@@ -601,6 +601,7 @@
       ENDDO
 
       CALL GLOBALSUM(grid, TEIJ, totalEnergy, ALL=.true.)
+      call ffdd_te(KEIJ,PEIJ,TEIJ,totalEnergy)
 
       end function getTotalEnergy
 
@@ -624,11 +625,13 @@
       I_1 = grid%I_STOP
 
       ediff = deltaEnergy / ((PSF-PMTOP)*SHA*mb2kg)
+      call ffdd_eadd(1,deltaEnergy,ediff)
 
       do l=1,lm
         T(I_0:I_1,J_0:J_1,L)=T(I_0:I_1,J_0:J_1,L)
      &       -ediff/PK(L,I_0:I_1,J_0:J_1)
       end do
+      call ffdd_eadd(2,deltaEnergy,ediff)
 
       end subroutine addEnergyAsDiffuseHeat
 
